### PR TITLE
[WIP] Enable type checking of edb.schema

### PR DESCRIPTION
**Pull request by ambv**
Each commit is a separate file covered by MonkeyType. I haven't reformatted it to satisfy flake8 so that the diff is as easy to review as possible.

MonkeyType covering new functions actually increases the number of Mypy errors for a while. I will be fixing those new errors in subsequent commits.

[work in progress]

### CODE REVIEW

File: edb/schema/annos.py [hunk 1]
```diff
@@ -31,6 +31,11 @@
 from . import utils
 
 
+if typing.TYPE_CHECKING:
```
Comment (1st1) [inline edb/schema/annos.py]: `from typing import *  # NoQA`?

Comment (ambv) [reply edb/schema/annos.py]: I have a script to refactor all those at a later point. Same what I used for object to typing.Any yesterday. I can apply that in this pull request in the last commit. For now I'd like the clarity of which Tuple is which, and so on.

Comment (1st1) [reply edb/schema/annos.py]: Sure!

Comment (1st1) [reply edb/schema/annos.py]: Sure!

**Comment by 1st1** [comment]
Since this is a WIP I'm not sure if you expect a review or comments at this point, but so far so good, all types appear to be correct!

**Comment by ambv** [comment]
Just wanted to show some code since I've been poking at all this for a while with no visible effect so far. I gave up on fully automated coverage but this is pretty good already. We'll upstream this to MonkeyType.

**Comment by ambv** [comment]
Aborted, will do separate PRs per file.

**Pull request closed without merging**